full-stack-practice/server/server.ts

```diff
@@ -0,0 +1,57 @@
+import 'dotenv/config';
+import pg from 'pg';
+import express from 'express';
+import { ClientError, errorMiddleware } from './lib/index.js';
+
+const db = new pg.Pool({
+  connectionString: process.env.DATABASE_URL,
+  ssl: {
+    rejectUnauthorized: false,
+  },
+});
+
+const app = express();
+app.use(express.json());
+
+app.get('/api/products', async (req, res, next) => {
+  try {
+    const sql = `
+      select *
+        from "products"
+        order by "productId"
+    `;
+    const result = await db.query(sql);
+    res.json(result.rows);
+  } catch (err) {
+    next(err);
+  }
+});
+
+app.get('/api/products/:productId', async (req, res, next) => {
+  try {
+    const { productId } = req.params;
+    if (!Number.isInteger(+productId)) {
+      throw new ClientError(400, 'productId needs to be a number');
+    }
+    const sql = `
+      select *
+      from "products"
+      where "productId" = $1
+    `;
+    const params = [productId];
+    const result = await db.query(sql, params);
+    const product = result.rows[0];
+    if (!product) {
+      return res.status(404).json({ error: 'product not found' });
```

Comment:
Do not return the 404 like this. Instead, `throw new ClientError(404, message);`, like you do on line 34.